crates/sui-framework-snapshot/src/lib.rs

```diff
@@ -4,20 +4,35 @@
 use serde::{Deserialize, Serialize};
 use std::collections::BTreeMap;
 use std::{fs, io::Read, path::PathBuf};
-use sui_framework::SystemPackage;
+use sui_framework::{BuiltInFramework, SystemPackage};
+use sui_protocol_config::ProtocolVersion;
 use sui_types::base_types::ObjectID;
-use sui_types::{
-    DEEPBOOK_PACKAGE_ID, MOVE_STDLIB_PACKAGE_ID, SUI_FRAMEWORK_PACKAGE_ID, SUI_SYSTEM_PACKAGE_ID,
-};
 
-pub type SnapshotManifest = BTreeMap<u64, SingleSnapshot>;
+const GIT_REVISION: &str = {
```

Comment:
That is correct, this cannot be present in a non-main.rs file otherwise there is a chance that it falls out of sync with the actual commit something is built from. please remove this and if you actually do need git revision information you'll need to require that it is passed in from an external source.

Reply:
specifically you can build the lib at revision X and then switch to a different commit and build a binary that depends on this lib at revision Y and the lib will report being built at X when you actually want the thing to report that it was built at revision Y